src/components/Main/Main.js

```diff
@@ -0,0 +1,393 @@
+import React from 'react';
+import Cards from '../Cards/Cards';
+import { BrowserRouter, Redirect, Route } from 'react-router-dom';
+import Menu from '../Menu/Menu';
+import WinScreen from '../WinScreen/WinScreen';
+import Settings from '../Settings/Settings';
+import Statistics from '../Statistics/Statistics';
+import buttonClickSound from '../../assets/audio/button-click-sound.mp3';
+import cardTurnSound from '../../assets/audio/card-turn.mp3';
+import victorySound from '../../assets/audio/victory.mp3';
+
+import './Main.scss';
+
+class Main extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      cardSet: this.props.cardSet,
+      cardSetType: this.props.cardSetType,
+      isGameFinish: false,
+      isMusicOn: false,
+      isSoundOn: false,
+      musicVolume: 0.5,
+      soundsVolume: 0.5,
+      moveCounter: 0,
+      language: 'en',
+      cardsQuantity: this.props.cardsQuantity,
+    };
+    this.firstTurnedCard = null;
+    this.secondTurnedCard = null;
+    this.isGameFinish = false;
+  }
+
+  isPlayerWon = () => {
+    return this.state.cardSet.every((cardObj) => cardObj.isGuessed === true);
+  };
+
+  setGameFinishState = () => {
+    this.setState({
+      isGameFinish: true,
+    });
+  };
+
+  onMusicSwitcherClick = () => {
+    const audio = document.querySelector('.game-music');
```

Comment:
you should use [ref](https://ru.reactjs.org/docs/refs-and-the-dom.html) instead of this